Clarify that assertions do not throw. Fixes #2624 (#2631)
avajs/ava: docs/03-assertions.md

});
```

> **Note**: Unlike other assertion libraries, AVA assertions do ***not***
> stop the test when they fail.
> 
> When a test has multiple assertion failures,
> AVA only displays the *first* failure.
> There's currently no way to see all failures from a test
> ([see issue #261](https://github.com/avajs/ava/issues/261)).

Reviewer: This should come at the end of the section, as it was before. The section starts with how you can access the assertions and how they're bound. An explanation of their runtime behavior shouldn't be sandwiched in between.

As for the text itself, what do you think about this?

```suggestion
AVA's assertions do not throw errors. Your test will continue executing after an assertion failure. AVA only displays the *first* failure.
```

## Built-in assertions

> **Note**: Unlike other assertion libraries, these assertions
> do ***not*** throw on failure.

Reviewer: Good call on repeating it here. With my suggested copy that would become:

```suggestion
AVA's assertions do not throw errors. Your test will continue executing after an assertion failure.
```

### `.pass(message?)`

Passing assertion.

### `.fail(message?)`

> Failing assertion. Does not throw.

Reviewer: I'd rather remove this. This describes how the specific assertion behaves, not sure it aids readability to repeat that the assertion never throws.
